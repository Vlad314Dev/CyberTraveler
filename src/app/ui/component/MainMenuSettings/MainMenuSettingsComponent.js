import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import {
    BACK_OPTION,
    optionLabels,
    SETTINGS_AUDIO,
    SETTINGS_SHOW_CONTROLS,
} from 'UIStore/MainMenu/MainMenuConfig';

class MainMenuOptionComponent extends PureComponent
{
    static propTypes = {
        activeOption: PropTypes.string.isRequired,
        setActiveOption: PropTypes.func.isRequired,
        userSettings: PropTypes.object.isRequired
    };

    setActiveOption = (optionName) => () => {
        const { setActiveOption } = this.props;
        setActiveOption(optionName);
    }

    getActiveOptionLabel()
    {
        const { activeOption } = this.props;
        const option = optionLabels.find((data) => data.key === activeOption);

        return option.label;
    }

    getOptionLabel(option)
    {
        if (option.key === SETTINGS_AUDIO) {
            const { userSettings } = this.props;
            const labelPostfix = userSettings.audio.noAudio ? 'Off' : 'On';
            
            return `${ option.label } - ${ labelPostfix }`
        }

        return option.label;
    }

    render()
    {
        return (
            <div block="MainMenuOption" elem="List">
                <span block="MainMenuOption" elem="ActiveLabel">{ this.getActiveOptionLabel() }</span>
                {optionLabels.map((option, index) => {
                    if (option.key === SETTINGS_AUDIO
                        || option.key === SETTINGS_SHOW_CONTROLS
                        || option.key === BACK_OPTION 
                    ) {
                        return <div 
                            block="MainMenuOption" 
                            elem="Wrapper" 
                            key={ index }
                        >
                            <span 
                                block="MainMenuOption" 
                                elem="Value"
                                key={ index }
                                onClick={ this.setActiveOption(option.key) }
                            >{ this.getOptionLabel(option) }</span>
                        </div>
                    }
                    
                    return null;
                })}
            </div>
        );
    }
}

export default MainMenuOptionComponent;
