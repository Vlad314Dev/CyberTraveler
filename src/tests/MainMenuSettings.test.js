/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import MainMenuSettingsComponent from '../app/ui/component/MainMenuSettings/MainMenuSettingsComponent';

describe('MainMenuSettings', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <MainMenuSettingsComponent 
                activeOption={ 'SETTINGS_OPTION' }
                setActiveOption={ jest.fn() }
                userSettings={ {
                    audio: {
                        noAudio: 0
                    }
                } }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <MainMenuSettingsComponent 
                activeOption={ 'SETTINGS_OPTION' }
                setActiveOption={ jest.fn() }
                userSettings={ {
                    audio: {
                        noAudio: 0
                    }
                }  }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
