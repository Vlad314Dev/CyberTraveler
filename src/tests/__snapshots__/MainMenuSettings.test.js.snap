// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuSettings Should render layout 1`] = `
Array [
  <div
    className="MainMenuOption__List"
  >
    <span
      className="MainMenuOption__ActiveLabel"
    >
      Settings
    </span>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Keyboard Controls
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Mobile Controls
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Audio - On
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Back
      </span>
    </div>
  </div>,
]
`;
